src/mixins/metadata.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import { addMetadata } from 'docc-render/utils/metadata';
+
+export default {
+  methods: {
+    /**
+     * Extract and join all the values from key 'text' on every object inside an array of objects,
+     * if there is an inlineContent array, it will extract all its text recursively as well
+     * @param {Array} content - array of objects containing "text" key
+     * @returns {String}
+     */
+    extractText(content) {
+      return content.reduce((acc, { text, inlineContent = [] }) => (
```

Comment:
Definitely not a blocker, but it might be nice to create a generic `ContentNode.reduce` function in the future that works on a content tree in the same way that we did for `map` and `forEach` so that we could write more of these kinds of functions without having to think about and repeat the recursive logic that's needed each time.

Reply:
I moved `extractText` inside ContentNode and added couple of tests for it: https://github.com/apple/swift-docc-render/pull/46/commits/79661b0867292c50c3cd4a6b990f53ea1cfb4a48

Reply:
I don't think I communicated what I meant with this comment well enough, sorry.

I just opened a PR that better captures my intent with this feedback: https://github.com/marinaaisa/swift-docc-render/pull/1

Please let me know what you think—I think it will be slightly safer for varying types of content and will also provide some generic building blocks for doing other transformations in the future if needed.